python/examples/minimal_build/build_conda.sh

```diff
@@ -28,8 +28,6 @@ CPP_BUILD_DIR=$HOME/arrow-cpp-build
 ARROW_ROOT=/arrow
 PYTHON=3.10
 
-git clone --depth=100 https://github.com/apache/arrow.git /arrow
-
 #----------------------------------------------------------------------
```

Comment:
Is this script used only in the minimal build docker images? If yes then could we use `ci/scripts/install_conda.sh` instead in the dockerfile?

Reply:
Yes, this is used only on the minimal build examples for conda. I am not entirely sure about using the `ci/scripts/install_conda.sh`. 
The current script does not only install conda but contains all the steps for the minimal build in order to help debug basic build environments, based on the `README.md`:
```
This directory shows how to bootstrap a local build from source on Linux with
an eye toward maximum portability across different Linux distributions. This
may help for contributors debugging build issues caused by their local
environments.
```
In my opinion splitting the `install/build` steps on two different scripts, the install on `ci/scripts/install_conda.sh` and the build on `python/examples/minimal_build/build_conda.sh` might be confusing for the purpose of debugging minimal builds.
I also understand that having two "install_conda" scripts is also not ideal.
Is your idea to have these two steps split on two different scripts?
If that's the case I would propose to include it on a different ticket.

Reply:
Sure, feel free to defer it to a follow-up ticket.

Installing conda can take significant time which is hard to iterate with. This time can be reduced by installing conda during docker-build in the relevant dockerfile. We could also switch to use the official conda image instead but still installing the conda dependencies from the dockerfile.